code/datums/elements/hat_wearer.dm

```diff
@@ -0,0 +1,73 @@
+/**
+ * # mobs that can wear hats!
+ */
+/datum/element/hat_wearer
+	element_flags = ELEMENT_DETACH_ON_HOST_DESTROY|ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	///offsets of hats we will wear
+	var/list/offsets
+
+/datum/element/hat_wearer/Attach(datum/target, offsets = list())
+	. = ..()
+	if (!isliving(target))
+		return ELEMENT_INCOMPATIBLE
+	src.offsets = offsets
+	RegisterSignal(target, COMSIG_ATOM_UPDATE_OVERLAYS, PROC_REF(on_overlays_updated))
+	RegisterSignal(target, COMSIG_ATOM_EXITED, PROC_REF(exited))
+	RegisterSignal(target, COMSIG_ATOM_ENTERED, PROC_REF(on_entered))
+	RegisterSignal(target, COMSIG_ATOM_ATTACKBY, PROC_REF(on_attack_by))
+
+/datum/element/hat_wearer/Detach(datum/target)
+	var/obj/item/hat = (locate(/obj/item/clothing/head) in target)
+	if(hat)
+		hat.forceMove(get_turf(target))
+	UnregisterSignal(target, list(
+		COMSIG_ATOM_UPDATE_OVERLAYS,
+		COMSIG_ATOM_EXITED,
+		COMSIG_ATOM_ENTERED,
+		COMSIG_ATOM_ATTACKBY,
+		COMSIG_ATOM_HITBY,
```

Comment:
We aren't registering HITBY why are we deregistering it

Reply:
op i was planning to make it so u can throw hats on them but i forgot to implement that